tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs

```diff
@@ -102,45 +144,149 @@ private void SetGeneralConfig(TracerSettings tracerSettings, DebuggerSettings se
+        {
+            return UpdateProductsState(tracerSettings, newDebuggerSettings ?? DebuggerSettings);
+        }
+
+        private Task UpdateProductsState(TracerSettings tracerSettings, DebuggerSettings newDebuggerSettings)
+        {
+            if (_processExit.Task.IsCompleted)
+            {
+                return _processExit.Task;
+            }
+
+            OneTimeSetup(tracerSettings);
+
+            lock (_syncLock)
+            {
+                if (_processExit.Task.IsCompleted)
+                {
+                    return _processExit.Task;
+                }
+
+                DebuggerSettings = newDebuggerSettings;
+                SetCodeOriginState(newDebuggerSettings);
+                SetExceptionReplayState(newDebuggerSettings);
+            }
+
+            return ScheduleStartDynamicInstrumentation(tracerSettings, newDebuggerSettings);
+        }
+
+        private void OneTimeSetup(TracerSettings tracerSettings)
+        {
+            if (Interlocked.CompareExchange(ref _initialized, 1, 0) != 0)
+            {
+                return;
+            }
+
+            LifetimeManager.Instance.AddShutdownTask(ShutdownTasks);
+            SetGeneralConfig(tracerSettings, DebuggerSettings);
+            InitializeSymbolUploader(tracerSettings);
+            if (tracerSettings.StartupDiagnosticLogEnabled)
+            {
+                _ = Task.Run(WriteStartupDebuggerDiagnosticLog);
+            }
+        }
+
+        private void InitializeSymbolUploader(TracerSettings tracerSettings)
         {
             try
             {
-                return TraceUtil.NormalizeTag(tracerSettings.ServiceName ?? TracerManager.Instance.DefaultServiceName);
+                if (_processExit.Task.IsCompleted || !DebuggerSettings.SymbolDatabaseUploadEnabled)
+                {
+                    return;
+                }
+
+                if (!tracerSettings.IsRemoteConfigurationAvailable)
+                {
+                    Log.Debug("Remote configuration is not available in this environment, so we don't upload symbols.");
+                    return;
+                }
+
+                var tracerManager = TracerManager.Instance;
+                SymbolsUploader = DebuggerFactory.CreateSymbolsUploader(tracerManager.DiscoveryService, RcmSubscriptionManager.Instance, ServiceName, tracerSettings, DebuggerSettings, tracerManager.GitMetadataTagsProvider);
+
+                _ = Task.Run(
+                    async () =>
+                    {
+                        try
+                        {
+                            await SymbolsUploader.StartFlushingAsync().ConfigureAwait(false);
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.Error(ex, "Failed to initialize symbol uploader");
+                        }
+                    });
```

Comment:
Why not 
```
.ContinueWith(
                         t => Log.Error(t?.Exception, "Failed to initialize symbol uploader"),
                         TaskContinuationOptions.OnlyOnFaulted);
```

Reply:
Fixed in dd64f6e89a16c1969abd9c1deb54d70d0f8c89c4